src/internal/operators/shareReplay.ts

```diff
@@ -108,13 +108,17 @@ function shareReplayOperator<T>({
           subject.complete();
         },
       });
+      if (isComplete) {
+        subscription = undefined;
+      }
     } else {
       innerSub = subject.subscribe(this);
     }
 
     this.add(() => {
       refCount--;
       innerSub.unsubscribe();
+      innerSub = undefined;
```

Comment:
This isn't necessary, really. As there would be no more references to `innerSub` held after this, and it would get GC'ed.